sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/util/CharVarcharUtils.scala

```diff
@@ -126,6 +126,16 @@ object CharVarcharUtils extends Logging {
     }
   }
 
+  /**
+   * Re-construct the original schema from the type string in the given metadata of each field.
+   */
+  def restoreCharVarCharInSchema(schema: StructType): StructType = {
```

Comment:
there is a `getRawType`, shall we name it `getRawSchema`?

Reply:
sgtm